### PR TITLE
[ci][test] exclude model download time in server start time

### PR DESCRIPTION
**Pull request by youkaichao**
fixes errors observed in https://buildkite.com/vllm/fastcheck/builds/3075#019182d5-5e41-408c-89e8-1d6bb86cc729

### CODE REVIEW

**Review by DarkLight1337 — APPROVED** [review]

**Comment by DarkLight1337** [comment]
We should have ran `entrypoints-tests` before merging this. Since the model name is changed, test cases that check the model name now fail.

File: tests/utils.py [hunk 1]
```diff
@@ -64,6 +65,10 @@ def __init__(self,
                  env_dict: Optional[Dict[str, str]] = None,
                  auto_port: bool = True,
                  max_wait_seconds: Optional[float] = None) -> None:
+        if not model.startswith("/"):
+            # download the model if it's not a local path
+            # to exclude the model download time from the server start time
+            model = snapshot_download(model)
```
Comment (mgoin) [inline tests/utils.py:71]: snapshot_download can download many more files than we need for inference, such as duplicate .pt files when we prefer to use safetensors. We should try to use a common function with how vLLM usually pulls down files for models

Comment (youkaichao) [reply tests/utils.py:71]: can you come up with a fix for it?

I don't know if safetensors would be enough. but we can have atry.

Comment (mgoin) [reply tests/utils.py:71]: Basically I am proposing using `DefaultModelLoader._prepare_weights`
https://github.com/vllm-project/vllm/blob/ea9fa160e3b47e0b8aa273f3eb2be410bd1ccab5/vllm/model_executor/model_loader/loader.py#L232
I can make a PR for this later today

Comment (youkaichao) [reply tests/utils.py:71]: please go ahead, I do see more files are downloaded in https://buildkite.com/vllm/fastcheck/builds/3094#01918525-98da-4596-8d31-f4e2c1172455

**Pull request merged**